### ISSUE TITLE
How to increase mAP for object detection in medical images 

### ISSUE THREAD

**Issue by mariojy10** [AlexeyAB/darknet#6406]
Hello @AlexeyAB  first of all i want to thank you for the great works and now i kinda need our suggestion.
Currently im working on **object detection problem in medical images**. To be specific im trying to detect 'bening' and 'malignant' masses (**2 class**) in mammogram (x-ray for breast). I have done about 9 training using this repository and of course following the readme. For the first eight training the result is pretty bad with the best model only got **mAP of **17%****  but evaluation on the trainset give mAP of 98%). 
I kinda know this gonna bound to happen cause my dataset is too small where i only have about **90 images for the training set** and **22 images for the test set** (80%,20% split ratio). In those first eight training i did an offline augmentation (yolo augmentation off ) so the trainset size become 4k - 6k. I use **the pretrain weight yolov4.conv.137**. 
For these experience i learned that maybe the test set is not 'match' with the train set or maybe i do too much augmentation but i tried to not much augment the trainset but still same low mAP. 

Now i managed to obtain new data and my data is looking like this :
![Number of Images per Class in Train set dan Test set MyDDSM (NoCALC)](https://user-images.githubusercontent.com/61630243/89286644-c3f31d00-d67c-11ea-9e51-5b15b8029fb5.png)
(FYI mix is images which have benign and malignant object -> i separate these mix images because i want to balance the class distributions when splitting)

Then I add more images for the trainset by doing 12x offline augmentation and train using yolov4 model with this cfg file :
[yolov4-cfg.txt](https://github.com/AlexeyAB/darknet/files/5021596/yolov4-cfg.txt)

The final result got mAP 30.3% (in the image its stated 30.2% cause its the last saved image before i stopped the training) and the best mAP is 34%. 
![chart_yolo-obj](https://user-images.githubusercontent.com/61630243/89286680-d705ed00-d67c-11ea-92d0-16b64c9b06f2.png)

My question is :
1. In this situation should i continue / finish the training ? Or im doing the correct things because the mAP no longer improve ...
2. Any suggestion for my cfg file ? - FYI im training on a laptop using RTX2060 6GB with Ubuntu 18.04 
3. Latter did i know this repo doesnt use letter_box activated when resizing images, in my dataset the images have varied resolutions, is it recommended to activate the letter_box ? [right now im training using lettex_box=1 with a liitle bit more data / i managed to scrap more images lol , will update you the result.]  
4. Any suggestion for improving mAP ? i want to atleast reach 85% mAP xDDD.
Thank you so much hope this question reach you.

```

**Comment by stephanecharette** [comment]
As you state, the number of images is so small that you wont get much out of it.  But when I'm setting up a NN and have just a few images to start with, I set the validation and the training file to be all of my images.  Especially if you only have ~90 images.  Then once the network is up, you can use that to markup more images.  Only way you'd get higher mAP results with such a small number of images is if they were nearly identical.

In the past, I found that images of different size have surprisingly little impact.  For example:
- https://www.ccoderun.ca/programming/ml/cancer_cells.html
- https://www.ccoderun.ca/programming/ml/dog_vs_cat.html

I know about the old letterbox flag, but truth is I've never tried it.  My understanding (and this may be flawed) is this used to be the standard in the original pjreddie/darknet repo, but it hasn't been that way in some time now.

How large are the areas with the tumours you are marking up to recognize compared to the original images?

**Comment by mariojy10** [comment]
> As you state, the number of images is so small that you wont get much out of it. But when I'm setting up a NN and have just a few images to start with, I set the validation and the training file to be all of my images. Especially if you only have ~90 images. Then once the network is up, you can use that to markup more images. Only way you'd get higher mAP results with such a small number of images is if they were nearly identical.
> 
> In the past, I found that images of different size have surprisingly little impact. For example:
> 
>     * https://www.ccoderun.ca/programming/ml/cancer_cells.html
> 
>     * https://www.ccoderun.ca/programming/ml/dog_vs_cat.html
> 
> 
> I know about the old letterbox flag, but truth is I've never tried it. My understanding (and this may be flawed) is this used to be the standard in the original pjreddie/darknet repo, but it hasn't been that way in some time now.
> 
> How large are the areas with the tumours you are marking up to recognize compared to the original images?

Hi thanks for sharing :)
1. For the first eight training event though i used small dataset i did an offline augmentation, but yes i knew that wont help much.
2. I used the letterbox flag because i want to preserve the aspect ratio of my images. I wanted the object not to be distorted when resize happen. By doing this i wanted the network not to falsely learn the distorted shape.
3. I yet measure the area / size of the object because there is to many of them. i only measure few of them when using the small dataset (first eight training) and the size is ok not too small  (below 1% image res). I think i will enable the small_object params for the next training just in case to avoid learning object that too small the model can handle.  
4. I did train a new model and now it reached mAP of 44.3%, i achieved this by adding about 60 images for each class (see the first pict). Then the rest is same.
